File: airpulse-ui/src/app/pages/admin/services/airport.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { BaseService } from "../../shared/services/base.service";
import { LoaderService } from "../../shared/services/loader.service";
import { AirportResponse, AirportRequest } from "../model/airport.model";

@Injectable({
    providedIn: 'root',
})
export class AirportService extends BaseService<AirportResponse, AirportRequest> {
    constructor(protected override http: HttpClient, protected override loader: LoaderService) {
        super(http, 'fc/airport', loader);
    }
}

File: flight-catalogue/flight-catalogue-model/src/main/java/com/pulse/air/flightcatalogue/contract/AirportService.java
package com.pulse.air.flightcatalogue.contract;

import com.pulse.air.common.model.BaseSearchRequest;
import com.pulse.air.commons.contract.BaseCRUDService;
import com.pulse.air.flightcatalogue.model.airport.AirportRequest;
import com.pulse.air.flightcatalogue.model.airport.AirportResponse;

public interface AirportService extends BaseCRUDService<AirportResponse, AirportRequest, BaseSearchRequest> {

}
